lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -621,6 +621,17 @@ SmallVector<StringAttr> standardOutDimNames(MLIRContext *ctx, int rank) {
   return ret;
 }
 
+// Returns ["dim0", "dim1", ..., "dim<rank-1>"] in given order.
+SmallVector<StringAttr> orderedOutDimNames(MLIRContext *ctx,
+                                           ArrayRef<unsigned> order) {
+  auto rank = order.size();
+  SmallVector<StringAttr> ret;
+  for (int i = 0; i < rank; i++) {
+    ret.push_back(StringAttr::get(ctx, "dim" + llvm::Twine(order[i])));
+  }
+  return ret;
+}
+
```

Comment:
This function is already somewhere else, could you move it from there?
cc @Mogball candidate to have in that file of LL utils

Reply:
It's `standardOutDimNames`.
Also see https://github.com/triton-lang/triton/pull/5470 for a nice place to put that utility function

Reply:
Ah, you wanted the dims permuted, yeah, use `permuteDimNames` and the standard one perhaps

Reply:
Yes, `combineCtaCgaWithShape` implicitly calculates order of repetitions from `ctaLayout` argument, so I transpose all cta components in this order with `transposeOuts`method.